projects/packages/publicize/src/rest-api/class-connections-controller.php

```
@@ -0,0 +1,431 @@
+			'can_refresh'   => 'userCanRefresh',
+			'refresh_text'  => 'refreshText',
+			'refresh_url'   => 'refreshURL',
+			'connection_id' => 'connectionID',
+		);
+
+		foreach ( $test_results as $test_result ) {
+			$result_fields = array(
+				// Compare to `true` because the API returns a 'must_reauth' for LinkedIn.
+				'status' => true === $test_result['connectionTestPassed'] ? 'ok' : 'broken',
+			);
+			foreach ( $mapping as $field => $test_result_field ) {
+				$result_fields[ $field ] = $test_result[ $test_result_field ];
+			}
+			$test_results_map[ $test_result['connectionID'] ] = $result_fields;
+		}
+
+		return $test_results_map;
+	}
+
+	/**
+	 * Get the connections data, either directly or by calling the WPCOM API
+	 *
+	 * @param bool $run_tests Whether to run the tests to check the health of the connections.
+	 *
+	 * @return array Connection objects in the same shape as from `get_connections`
+	 */
+	public static function get_connections_data( $run_tests = false ) {
+		// TODO decide on caching
+		return ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ? self::get_connections( $run_tests ) : self::get_connections_from_wpcom( $run_tests );
+	}
+
+	/**
+	 * Proxy the request to the WPCOM API
+	 *
+	 * @param bool $run_tests Whether to run the tests to check the health of the connections.
+	 *
+	 * @return array Connection objects in the same shape as from `get_connections`
+	 */
+	protected static function get_connections_from_wpcom( $run_tests = false ) {
+		$site_id = Manager::get_site_id( true );
+		if ( ! $site_id ) {
+			return array();
+		}
+
+		$path = add_query_arg(
+			array( 'test_connections' => $run_tests ),
+			sprintf( '/sites/%d/publicize/connections', $site_id )
+		);
+
+		$response = Client::wpcom_json_api_request_as_user( sprintf( $path, $site_id ), 'v2', array( 'method' => 'GET' ) );
+
+		if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			// TODO log error.
```

I'm trying to work out what's best to do here. I feel like we should return an error. If it's empty like this then everything will still function but will be blank. How do we get an error message to the user? Maybe it's ok for now

---

We should always return an array here and pass the error in some other way like JITM.